doc/design/wallet/wallet-design.md

```diff
@@ -0,0 +1,762 @@
+                                           /// credentials: wallet storage credentials (For example, database credentials)
+                                           /// storage_handle_p: pointer to store opened storage handle
+                                           open: Option<extern fn(name: *const c_char,
+                                                                  config: *const c_char,
+                                                                  runtime_config: *const c_char,
+                                                                  credentials: *const c_char,
+                                                                  storage_handle_p: *mut i32) -> ErrorCode>,
+
+                                           /// Close the opened walled storage (For example, closing database connection)
+                                           ///
+                                           /// #Params
+                                           /// storage_handle: opened storage handle (See open handler)
+                                           close: Option<extern fn(handle: i32) -> ErrorCode>,
+
+                                           /// Create a new record in the wallet storage
+                                           ///
+                                           /// #Params
+                                           /// storage_handle: opened storage handle (See open handler)
+                                           /// type_: allows to separate different record types collections
+                                           /// id: the id of record
+                                           /// value: the value of record (pointer to buffer)
+                                           /// value_len: the value of record (buffer size)
+                                           /// tags_json: the record tags used for search and storing meta information as json:
+                                           ///   {
+                                           ///     "tagName1": "tag value 1", // string value
+                                           ///     "tagName2": 123, // numeric value
+                                           ///   }
+                                           ///   Note that null means no tags
+                                           add_record: Option<extern fn(storage_handle: i32,
+                                                                        type_: *const c_char,
+                                                                        id: *const c_char,
+                                                                        value: *const u8,
+                                                                        value_len: usize,
+                                                                        tags_json: *const c_char) -> ErrorCode>,
```

Comment:
We also need to transfer value_key (encrypted key which is used for encrypting the value). Every value have its own key.
It is the same for all functions transferring the value.